event_callback.go

```diff
@@ -0,0 +1,170 @@
+package firego
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/zabawaba99/firego/sync"
+)
+
+// ChildEventFunc is the type of function that is called for every
+// new child added under a firebase reference. The snapshot argument
+// contains the data that was added. The previousChildKey argument
+// contains the key of the previous child that this function was called for.
+type ChildEventFunc func(snapshot DataSnapshot, previousChildKey string)
```

Comment:
When using this pattern, I've seen a lot of packages name their func type something like `FooEventHandler` as opposed to `FooEventFunc`. What do you think about making this `ChildEventHandler`?